rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -245,6 +264,38 @@ public Topology getTopology() {
         }
     }
 
+    private NodeSet getNodeSetWithExtraNodes(Set<Node> nodes, int number) {
+        Set<Node> main = new LinkedHashSet<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "main" object should be an interface such as "Set" rather than the implementation "LinkedHashSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)